### ISSUE TITLE
Link checker should check .xml files as well

### ISSUE THREAD

**Issue by heaths** [Azure/azure-sdk-tools#6107]
.NET uses .xml files for documentation, and we recently discovered some have broken (404) links. We should add .xml files to the link checker as well or, if too noisy for all language repos, make the file pattern list configurable per-repo.

See Azure/azure-sdk-for-net#36066 for context.

**Comment by heaths** [comment]
@maririos @weshaggard I checked for various possible labels beyond this, but nothing stood out. Sorry. Let me know if I can help further triaging this.

**Comment by weshaggard** [comment]
I assume this is related to https://github.com/Azure/azure-sdk-for-net/issues/22216. Currently we don't have a good way to handle that because it takes some effort to figure out what is a valid link or not. I was thinking we could actually check the generated docs for broken links but no one has been able to look into that yet.

**Comment by heaths** [comment]
Similar, but not specifically related to that issue, yes.

> Currently we don't have a good way to handle that because it takes some effort to figure out what is a valid link or not.

@weshaggard I just mean URLs, which we already have logic for. It even handles PRs that add links to `main`, for which I opened a bug long ago and was fixed shortly thereafter. I don't mean links to other types. So a regex-based search and "will it target main once merged" check - which we already use - should suffice.

**Comment by weshaggard** [comment]
Yes we have support for validating urls but we don't have support for finding url's to validate for anything outside of html or MD. So the issue is adding support for finding the url's in new file types as we don't want to blindly look for anything that may be a url in every text file as that will end up with a lot of false positives.
